tools/Common/IceRpc.BuildTelemetry/TelemetryTask.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) ZeroC, Inc.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace IceRpc.BuildTelemetry;
+
+/// <summary>
+/// A custom MSBuild task to run telemetry commands.
+/// </summary>
+public class TelemetryTask : ToolTask
```

Comment:
I would create separate tasks for (in) IceRpc.Protobuf.Tools and IceRpc.Slice.Tools - simpler and cleaner. Same for the actual client code.

Reply:
In the interest of moving forward with this I think the current structure is fine. It does its job and is clean enough. This code is going to be refactored once we have the plugin system in the future so I don't want to spend too much time obsessing over having the cleanest possible telemetry client. It should be treat as an "Alpha" version of the telemetry at this point I think.

Reply:
I don't see the difficulty in splitting this task into two tasks. It should be a matter of a couple minutes. And it would make the code simpler since you can eliminate Idl.

This common task ties the Slice compilation and the Protobof compilation, which I don't think is desirable.